diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index e494d16ffce..17d6e07172e 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1152,6 +1152,7 @@ dtb-$(CONFIG_ARCH_IMX8M) += \
 	imx8mn-venice.dtb \
 	imx8mn-venice-gw7902.dtb \
 	imx8mq-evk.dtb \
+	fsl-imx8mq-ecu150a1.dtb \
 	imx8mm-beacon-kit.dtb \
 	imx8mn-beacon-kit.dtb \
 	imx8mq-mnt-reform2.dtb \
diff --git a/board/freescale/imx8mq_evk/Makefile b/board/freescale/imx8mq_evk/Makefile
index fd260bb1ec4..e70ab9ba5d5 100644
--- a/board/freescale/imx8mq_evk/Makefile
+++ b/board/freescale/imx8mq_evk/Makefile
@@ -8,9 +8,5 @@ obj-y += imx8mq_evk.o
 
 ifdef CONFIG_SPL_BUILD
 obj-y += spl.o
-ifdef CONFIG_IMX8MQ_4GB_DDR_TIMING
-obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_4g.o
-else
-obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_b0.o lpddr4_timing.o
-endif
+obj-$(CONFIG_IMX8M_LPDDR4) += lpddr4_timing_1g.o lpddr4_timing_2g.o lpddr4_timing_4g.o lpddr4_timing_b0.o
 endif
diff --git a/board/freescale/imx8mq_evk/imx8mq_evk.c b/board/freescale/imx8mq_evk/imx8mq_evk.c
index 607170e2d2c..d4c326249f4 100644
--- a/board/freescale/imx8mq_evk/imx8mq_evk.c
+++ b/board/freescale/imx8mq_evk/imx8mq_evk.c
@@ -33,7 +33,7 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
-#define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1)
+#define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1 | PAD_CTL_PUE)
 
 #define WDOG_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_HYS | PAD_CTL_PUE)
 
@@ -300,6 +300,9 @@ int board_late_init(void)
 	board_late_mmc_env_init();
 #endif
 
+	puts("DeviceName is ECU150A1.\n");
+	env_set("devicename", "ecu150a1");
+	env_set("bootargs_ram_capacity", "cma=384M");
 	return 0;
 }
 
diff --git a/board/freescale/imx8mq_evk/spl.c b/board/freescale/imx8mq_evk/spl.c
index c7775c326fc..9efeaa11b51 100644
--- a/board/freescale/imx8mq_evk/spl.c
+++ b/board/freescale/imx8mq_evk/spl.c
@@ -53,7 +53,7 @@ static void spl_dram_init(void)
 #else
 	/* ddr init */
 	if (soc_rev() >= CHIP_REV_2_1)
-		ddr_init(&dram_timing);
+		ddr_init(&dram_timing_4g);
 	else
 		ddr_init(&dram_timing_b0);
 #endif
@@ -234,6 +234,14 @@ void spl_board_init(void)
 			printf("\nsec_init failed!\n");
 	}
 
+#ifndef CONFIG_SPL_USB_SDP_SUPPORT
+	/* Serial download mode */
+	if (is_usb_boot()) {
+		puts("Back to ROM, SDP\n");
+		restore_boot_params();
+	}
+#endif
+
 	init_usb_clk();
 
 	puts("Normal Boot\n");
diff --git a/common/autoboot.c b/common/autoboot.c
index 1752dd58483..e00b75d3e78 100644
--- a/common/autoboot.c
+++ b/common/autoboot.c
@@ -43,6 +43,136 @@ DECLARE_GLOBAL_DATA_PTR;
 static int stored_bootdelay;
 static int menukey;
 
+static int advupdate(const char *ifname, const char *dev_part, const char *s)
+{
+	int ret = 0;
+	char buf[256] = {0};
+	bool haveconfig = true;
+
+	//start usb device
+	if (!strcmp(ifname,"usb"))
+	{
+		run_command("usb start", 0);
+	}
+
+	else if (!strcmp(ifname, "mmc"))
+	{
+		//load config file
+		env_set_ulong("filesize",0);
+		sprintf(buf, "fatload %s %s ${loadaddr} ${advufile};env import -t ${loadaddr} ${filesize}", ifname, dev_part);
+		run_command_list(buf, -1, 0);
+		if (env_get_ulong("filesize", 16, 0) == 0) {
+			memset(buf, 0, sizeof(buf));
+			sprintf(buf, "ext2load %s %s ${loadaddr} ${advufile};env import -t ${loadaddr} ${filesize}", ifname, dev_part);
+			run_command_list(buf, -1, 0);
+			if (env_get_ulong("filesize", 16, 0) == 0) {
+				// config file not found!
+				//printf("%s %s config file not found!\n", ifname, dev_part);
+				haveconfig = false;
+			}
+		}
+
+		// find update.bin file
+		if (!haveconfig)
+		{
+			memset(buf, 0, sizeof(buf));
+			sprintf(buf, "if test -e %s %s update.bin; then setenv advupdatebin y; fi", ifname, dev_part);
+			run_command_list(buf, -1, 0);
+			if (env_get_yesno("advupdatebin") != 1)
+			{
+				// update.bin file not found!
+				//printf("%s %s update bin file not found!\n", ifname, dev_part);
+				return 0;
+			}
+		}
+	}
+	else if (!strcmp(ifname, "nand"))
+	{
+	memset(buf, 0, sizeof(buf));
+	sprintf(buf, "ubi part NAND.recovery && ubifsmount ubi0;ubifsload ${loadaddr} ${advufile};env import -t ${loadaddr} ${filesize}");
+	run_command_list(buf, -1, 0);
+		run_command_list("ubifsumount", -1, 0);
+		if (env_get_ulong("filesize", 16, 0) == 0) {
+			// config file not found!
+			//printf("nand config file not found!!\n");
+			haveconfig = false;
+		}
+
+		// find update.bin file
+		if (!haveconfig)
+		{
+			memset(buf, 0, sizeof(buf));
+			sprintf(buf, "ubi part NAND.recovery && ubifsmount ubi0;ubifsload ${loadaddr} update.bin 0x10;env import -t ${loadaddr} 0x10");
+			run_command_list(buf, -1, 0);
+			run_command_list("ubifsumount", -1, 0);
+			if (env_get_ulong("filesize", 16, 0) == 0)
+			{
+				// update.bin file not found!
+				//printf("nand update bin file not found!!\n");
+				return 0;
+			}
+			else
+			{
+				run_command("setenv advupdatebin y", 0);
+			}
+		}
+	}
+
+	if (haveconfig)
+		printf("Config file found!\n");
+	else if (env_get_yesno("advupdatebin") == 1)
+		printf("update bin file found!\n");
+	else
+		return 0;
+
+	if ((env_get_ulong("filesize",16,0) > 0) || (env_get_yesno("advupdatebin") == 1))
+	{
+#if 1 //+= ramdisk boot
+		if ((env_get_yesno("advimage") == 1) || (env_get_yesno("advrootfs") == 1) || (env_get_yesno("advrecover") == 1) || (env_get_yesno("advupdatebin") == 1))
+		{
+			ret = 2;
+			printf("System recovery mode!\n");
+			if (!strcmp(ifname, "mmc"))
+			{
+				if (env_get_yesno("advupdatebin") == 1)
+				{
+					run_command("setenv mmcdev 0", 0);
+				}
+				else
+				{
+					if (!strcmp(dev_part, "1:1"))
+					{
+						run_command("setenv mmcdev 1", 0);
+					}
+					else if (!strcmp(dev_part, "0:3"))
+					{
+						run_command("setenv mmcdev 0", 0);
+					}
+				}
+				run_command("mmc dev ${mmcdev}", 0);
+
+				s = env_get("advrfs");
+				run_command_list(s, -1, 0);
+			}
+			else if (!strcmp(ifname, "usb"))
+			{
+				s = env_get("advrfs");
+				run_command_list(s, -1, 0);
+			}
+		}
+		else
+		{
+			printf("System boot mode!\n");
+			s = env_get("bootcmd");
+		}
+#endif
+	}
+	else
+		printf("Normal boot mode!\n");
+
+	return ret;
+}
+
 #if defined(CONFIG_AUTOBOOT_STOP_STR_CRYPT)
 #define AUTOBOOT_STOP_STR_CRYPT	CONFIG_AUTOBOOT_STOP_STR_CRYPT
 #else
@@ -381,9 +511,11 @@ static int abortboot_single_key(int bootdelay)
 	 * Check if key already pressed
 	 */
 	if (tstc()) {	/* we got a key press	*/
-		getchar();	/* consume input	*/
+		if (getchar() == 0x0d) // got enter key
+		{
 		puts("\b\b\b 0");
 		abort = 1;	/* don't auto boot	*/
+		}
 	}
 
 	while ((bootdelay > 0) && (!abort)) {
@@ -392,6 +524,8 @@ static int abortboot_single_key(int bootdelay)
 		ts = get_timer(0);
 		do {
 			if (tstc()) {	/* we got a key press	*/
+				if (getchar() == 0x0d) // got enter key
+				{
 				int key;
 
 				abort  = 1;	/* don't auto boot	*/
@@ -400,6 +534,7 @@ static int abortboot_single_key(int bootdelay)
 				if (IS_ENABLED(CONFIG_AUTOBOOT_USE_MENUKEY))
 					menukey = key;
 				break;
+				}
 			}
 			udelay(10000);
 		} while (!abort && get_timer(ts) < 1000);
@@ -523,6 +658,13 @@ void autoboot_command(const char *s)
 		if (lock)
 			prev = disable_ctrlc(1); /* disable Ctrl-C checking */
 
+		{
+			int ret = 0;
+			ret = advupdate("mmc", "1:1", s);
+			if (!ret)
+				ret = advupdate("mmc", "0:3", s);
+		}
+
 		run_command_list(s, -1, 0);
 
 		if (lock)
diff --git a/configs/imx8mq_evk_defconfig b/configs/imx8mq_evk_defconfig
index 36715241505..092cd2a65af 100644
--- a/configs/imx8mq_evk_defconfig
+++ b/configs/imx8mq_evk_defconfig
@@ -15,7 +15,7 @@ CONFIG_SYS_I2C_MXC_I2C2=y
 CONFIG_SYS_I2C_MXC_I2C3=y
 CONFIG_DM_GPIO=y
 CONFIG_SPL_TEXT_BASE=0x7E1000
-CONFIG_USB_TCPC=y
+#CONFIG_USB_TCPC=y
 CONFIG_TARGET_IMX8MQ_EVK=y
 CONFIG_DM_RESET=y
 CONFIG_SPL_MMC=y
@@ -25,7 +25,7 @@ CONFIG_SPL_STACK=0x187ff0
 CONFIG_SPL=y
 CONFIG_IMX_BOOTAUX=y
 CONFIG_SYS_LOAD_ADDR=0x40400000
-CONFIG_DEFAULT_DEVICE_TREE="imx8mq-evk"
+CONFIG_DEFAULT_DEVICE_TREE="fsl-imx8mq-ecu150a1"
 CONFIG_REMAKE_ELF=y
 CONFIG_BOOTCOMMAND="run sr_ir_v2_cmd;run distro_bootcmd;run bsp_bootcmd"
 CONFIG_SYS_MONITOR_LEN=524288
@@ -34,7 +34,8 @@ CONFIG_FIT_EXTERNAL_OFFSET=0x3000
 CONFIG_SPL_LOAD_FIT=y
 CONFIG_DISTRO_DEFAULTS=y
 CONFIG_OF_SYSTEM_SETUP=y
-CONFIG_SPL_MAX_SIZE=0x26000
+#CONFIG_SPL_MAX_SIZE=0x26000
+CONFIG_SPL_MAX_SIZE=0x27000
 CONFIG_SPL_HAS_BSS_LINKER_SECTION=y
 CONFIG_SPL_BSS_START_ADDR=0x180000
 CONFIG_SPL_BSS_MAX_SIZE=0x2000
@@ -56,8 +57,9 @@ CONFIG_SYS_PBSIZE=1050
 CONFIG_SPL_USB_GADGET=y
 CONFIG_SPL_USB_SDP_SUPPORT=y
 # CONFIG_BOOTM_NETBSD is not set
-# CONFIG_CMD_EXPORTENV is not set
-# CONFIG_CMD_IMPORTENV is not set
+CONFIG_CMD_EXPORTENV=y
+CONFIG_CMD_IMPORTENV=y
+CONFIG_BOOTDELAY=1
 CONFIG_CMD_NVEDIT_EFI=y
 CONFIG_CMD_CRC32=y
 CONFIG_CRC32_VERIFY=y
@@ -182,10 +184,11 @@ CONFIG_EFI_VAR_BUF_SIZE=139264
 CONFIG_EFI_IGNORE_OSINDICATIONS=y
 CONFIG_EFI_CAPSULE_AUTHENTICATE=y
 CONFIG_EFI_CAPSULE_ESL_FILE="CRT.esl"
-CONFIG_OPTEE=y
-CONFIG_CMD_OPTEE_RPMB=y
+#CONFIG_OPTEE=y
+#CONFIG_CMD_OPTEE_RPMB=y
 CONFIG_EFI_MM_COMM_TEE=y
-CONFIG_TEE=y
+#CONFIG_TEE=y
 CONFIG_EFI_ESRT=y
 CONFIG_EFI_HAVE_CAPSULE_UPDATE=y
-CONFIG_FIT_SIGNATURE=y
+#CONFIG_FIT_SIGNATURE=y
+CONFIG_SMBIOS_PARSER=y
diff --git a/include/configs/imx8mq_evk.h b/include/configs/imx8mq_evk.h
index 7b308748f90..759347e123d 100644
--- a/include/configs/imx8mq_evk.h
+++ b/include/configs/imx8mq_evk.h
@@ -11,6 +11,14 @@
 #include <asm/arch/imx-regs.h>
 #include "imx_env.h"
 
+#define ADV_IMX8_DDR_1G		0x0
+#define ADV_IMX8_DDR_2G     0x1
+#define ADV_IMX8_DDR_4G     0x2
+
+#define TEE_LOAD_ADDR_1G 	0x7e000000
+#define TEE_LOAD_ADDR_2G 	0x7e000000
+#define TEE_LOAD_ADDR_4G 	0xfe000000
+
 #ifdef CONFIG_SPL_BUILD
 /*#define CONFIG_ENABLE_DDR_TRAINING_DEBUG*/
 
@@ -39,6 +47,8 @@
 #define BOOTENV
 #endif
 
+#define CONSOLE					"console=ttymxc0,115200 earlycon=ec_imx6q,0x30860000,115200"
+
 /*
  * Another approach is add the clocks for inmates into clks_init_on
  * in clk-imx8mq.c, then clk_ingore_unused could be removed.
@@ -79,12 +89,12 @@
 	"bsp_script=boot.scr\0" \
 	"image=Image\0" \
 	"splashimage=0x50000000\0" \
-	"console=ttymxc0,115200\0" \
+	" "CONSOLE" \0" \
 	"fdt_addr_r=0x43000000\0"			\
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
-	"fdtfile=imx8mq-evk.dtb\0" \
+	"fdtfile=fsl-imx8mq-ecu150a1.dtb\0" \
 	"bootm_size=0x10000000\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=1\0" \
@@ -94,8 +104,8 @@
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bsp_script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr_r} ${fdtfile}\0" \
+	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
+	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr_r} ${fdtfile}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -107,7 +117,34 @@
 		"else " \
 			"echo wait for boot; " \
 		"fi;\0" \
-	"netargs=setenv bootargs ${jh_clk} ${mcore_clk} console=${console} " \
+	"emmcboot=echo Booting from emmc ...; setenv mmcdev 0; setenv mmcroot /dev/mmcblk0p2 rootwait rw;" \
+		"run mmcargs; " \
+		"mmc dev ${mmcdev};" \
+		"if run loadimage; then " \
+			"if run loadfdt; then " \
+				"booti ${loadaddr} - ${fdt_addr_r}; " \
+			"fi; " \
+		"fi;\0" \
+	"advufile=advupdate.txt\0" \
+	"ramrootfstype=ext2 rootwait\0" \
+	"loadramdisk=load mmc ${mmcdev} ${initrd_addr} ramdisk.gz\0" \
+	"advrargs=setenv bootargs console=${console} " \
+		"${optargs} " \
+		"root=/dev/ram0 rw ramdisk_size=65536 "\
+		"initrd=${initrd_addr},64M " \
+		"rootfstype=${ramrootfstype}\0" \
+	"advrfs=echo Advantech recovery file system ramdisk ...; " \
+		"setenv devnum ${mmcdev}; " \
+		"setenv devtype mmc; " \
+		"run loadimage; " \
+		"run loadramdisk; " \
+		"run advrargs; " \
+		"if run loadfdt; then " \
+			"booti ${loadaddr} - ${fdt_addr_r}; " \
+		"else " \
+			"booti ${loadaddr};" \
+		"fi\0" \
+	"netargs=setenv bootargs ${jh_clk} console=${console} " \
 		"root=/dev/nfs " \
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
 	"netboot=echo Booting from net ...; " \
@@ -137,7 +174,7 @@
 				   "else run netboot; " \
 				   "fi; " \
 			   "fi; " \
-		   "fi;"
+		   "else run emmcboot; fi;"
 
 /* Link Definitions */
 
@@ -147,9 +184,7 @@
 
 #define CFG_SYS_SDRAM_BASE           0x40000000
 #define PHYS_SDRAM                      0x40000000
-#define PHYS_SDRAM_SIZE			0xC0000000 /* 3GB DDR */
-#define PHYS_SDRAM_2			0x100000000
-#define PHYS_SDRAM_2_SIZE		0x40000000 /* 1GB */
+#define PHYS_SDRAM_SIZE			SZ_4G /* 4GB DDR */
 
 #define CFG_MXC_UART_BASE		UART_BASE_ADDR(1)
 
